ops/jujucontext.py

```diff
@@ -12,36 +12,84 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""A helper to work with the Juju context and version."""
+"""A helper to work with the Juju context."""
 
 from __future__ import annotations
 
 import dataclasses
+import os
 from pathlib import Path
 from typing import Any, Mapping
 
 from .jujuversion import JujuVersion
 
 
-@dataclasses.dataclass(frozen=True)
-class _JujuContext:
-    """_JujuContext collects information from environment variables named 'JUJU_*'.
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class JujuContext:
+    """JujuContext provides the Juju hook context.
 
-    Source: https://documentation.ubuntu.com/juju/3.6/reference/hook/#hook-execution.
-    The HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398.
-    Only a subset of the above source, because these are what are used in ops.
+    Juju provides context for the hook in the form of environment variables.
+    Rather than directly accessing the environment, charms should use
+    :meth:`ops.JujuContext.from_environ` to access this information.
+
+    Note that most of the information in ``JujuContext`` is exposed through the
+    framework, for example :attr:`ops.JujuContext.model_name` is
```

Comment:
Very minor suggestions:

```suggestion
    Most of the information in ``JujuContext`` is exposed through the
    framework. For example :attr:`ops.JujuContext.model_name` is
```